### PR TITLE
Add UniformNoise layer, and test

### PR DESCRIPTION
**Pull request by RitwikGupta**
Adding uniform noise to data is a common regularization technique that is currently missing from Keras. This is a simple layer that introduces that functionality.

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. Can you provide a few paper references for the use of this regularization method?

**Comment by RitwikGupta** [comment]
@fchollet Certainly.

- "Deep Learning is Robust to Massive Label Noise" from CSAIL uses uniform noise on labels as a way to show the robustness of deep models: https://arxiv.org/pdf/1705.10694.pdf
- "A Review Paper: Noise Models in Digital Image Processing" from the Indian Ministry of Defense reviews various types of noise, of which is is quantization noise, which follows a uniform distribution, and as such is referred to as uniform noise: https://arxiv.org/pdf/1505.03489.pdf
- Adding white uniform noise is also a commonly asked question on MATLAB forums: https://www.mathworks.com/matlabcentral/answers/69194-how-to-add-uniform-noise-to-an-image-using-imnoise-is-there-defind-description-like-gaussian-sp

**Comment by fchollet** [comment]
Thanks for the references! However, it appears they do not support the premise:

> Adding uniform noise to data is a common regularization technique

Merging new features involves estimating whether the usefulness (how badly users need it times how many people will use it) of the feature outweighs its cost (how much code and API surface gets added, and how complex it is). "Cost" means both maintenance cost (integrated over the total lifetime of the framework) and cognitive load imposed on users (also integrated).

The tradeoff looks negative for this feature right now. Unless you can make a better case, I'll close the PR.

**Comment by RitwikGupta** [comment]
@fchollet You may also want to look at this paper: https://jivp-eurasipjournals.springeropen.com/articles/10.1186/s13640-017-0188-z which uses various noises, including uniform noise, to train a robust face recognition CNN.
There are also various project which manually input uniform noise, such as: https://github.com/roatienza/Deep-Learning-Experiments/blob/master/Experiments/Tensorflow/GAN/dcgan_mnist.py#L157

Short of an extensive literature review, I think these two examples should be a convincing argument towards a UniformNoise layer. I completely agree with your cost-benefit view, however, so if it still doesn't make sense to include, please feel free to close the PR 😄

**Comment by fchollet** [comment]
Thanks for the further references. In these cases, this is input noise, which in Keras could be added on the fly via an input generator.

Overall it doesn't seem like additive uniform noise is being used as a regularization layer (as a replacement for Dropout). I'll close the PR.
